### DISCUSSION TITLE
How do I read the status?

### DISCUSSION THREAD

**Discussion by TAWood** [mvp/uhubctl#537] [Q&A] [answered]
I have tried to read the man page (here: https://manpages.ubuntu.com/manpages/focal/en/man8/uhubctl.8.html) as well as this github site and I can't seem to find an explanation of how to read the status codes reported for each port.

For example, I ran "sudo uhubctl" and the output contains these lines (I've just grabbed several examples, eliminating many duplicates in my actual output.)

  Port 1: 0100 power
  Port 1: 02a0 power 5gbps Rx.Detect
  Port 8: 0103 power enable connect 
  Port 4: 0263 power 5gbps U3 enable connect
  Port 4: 0503 power highspeed enable connect

  How do I read the 4 digit hex value?  What does each bit mean?

  Now that I've typed out this question it seems like the various words after the hex value are probably the decoded bits.  As I can see that the number of words after the hex bits is the same as the number of "on" bits in the hex value.  But is there some explanation for what all 16 bits are and what each bit means?

**Comment by mvp** [comment, marked as answer]
Status hexadecimal value is printed by [print_port_status](https://github.com/mvp/uhubctl/blob/f37dcb5cbfc/uhubctl.c#L722). It is simply low level USB descriptor value. Each bit is decoded and printed in human readable form next to it.
